ement-room.el

```diff
@@ -2419,6 +2419,59 @@ these all require at least version 29 of Emacs):
                                ('remove "removed from"))
                              (ement--format-room space))))))
 
+;;;;; Admin/power commands
+
+(defun ement-room--kick-ban-interactive (prompt)
+  "Interactive arguments for kicking or un/banning a user.
+See `ement-room-kick-user', `ement-room-ban-user', `ement-room-unban-user'.
+Returns (user room session reason)."
+  (ement-room-with-highlighted-event-at (point)
+    (let* ((room ement-room)
+           (event (ewoc-data (ewoc-locate ement-ewoc)))
+           (user (ement-event-sender event)))
+      (if (yes-or-no-p (format "%s user %s <%s> from room %s? "
+                               prompt
+                               (ement-user-displayname user)
+                               (ement-user-id user)
+                               (ement--format-room room)))
+          (list user ement-room ement-session
+                (read-string "Reason (optional): "
+                             nil nil nil 'inherit-input-method))
+        ;; HACK: This isn't really an error, but is there a cleaner way to cancel?
+        (user-error "Aborted")))))
```

Comment:
Looking at, e.g. `isearch-cancel`, it looks like maybe we should use `(signal 'quit nil)` (which I didn't realize before).  What do you think?

Reply:
I'm unsure about that.

`keyboard-quit` does this:

```el
  (let ((debug-on-quit nil))
    (signal 'quit nil))
```

which seems like what we'd want as well, as we're not triggering this on account of `C-g`.

But then, that's more or less the point of `user-error`:

```el
(defun user-error (format &rest args)
  "Signal a user error, making a message by passing ARGS to `format-message'.
This is like `error' except that a user error (or \"pilot error\") comes
from an incorrect manipulation by the user, not from an actual problem.
In contrast with other errors, user errors normally do not cause
entry to the debugger, even when `debug-on-error' is non-nil.
This can be overridden by `debug-ignored-errors'.
```

My initial reaction is to keep using `user-error`.

Reply:
My only hesitation is that pressing C-g isn't really an "incorrect manipulation", so I'm reluctant to use `user-error` and imply that the user did something wrong.  So I think I'd prefer to use `quit`.  (And I should make the same change in other relevant places now that I've realized it.)